actors.md

```diff
@@ -1109,8 +1150,35 @@ type PaymentChannel struct {
 	minCloseHeight UInt
 
 	laneStates {UInt:LaneState}
+	
+	verifActor Address
+	verifMethod Uint
```

Comment:
Yeah, I initially put these in the actor state on chain because I was concerned about increasing the cost of a miner cashing in their payment channel in the case where things arent quite right. But in reality, an address and a method ID are like, 6 extra bytes or so (ID address and a one byte method ID), so its probably worth the benefit of being able to have one payment channel be able to work with a variety of different verifiers.

Now, for the other thing you said:

> with N-1 vouchers that require verification and 1 at the end that doesn't. If a miner manages to get to the end of a deal without having their payment channel challenged, they get to collect without the expense of proving inclusion.

This is tricky, and opens up weird holes. and we probably shouldn't do it. If the client stops paying attention, not only can the miner avoid getting slashed for dropping the data, they could receive all the payment at the end of the contract.

Reply:
If the client has reached the end of the storage deal without noticing whether the miner has been storing their piece, does it matter whether the miner was actually storing it or not? What you say about the miner getting away with cheating is true, but the problem doesn't sound bad to me. A rational miner would have to assume the client is checking.

Reply:
Theres a big difference here. In one case, the miner just doesnt get their storage collateral slashed, and the client also doesnt lose any money. In the second case, the miner doesnt get its storage collateral slashed *and* they get to take the clients payment. This is a significantly worse failure mode.

Reply:
Ok, that's what I get for trying to put on a @whyrusleeping hat when trading off bandwidth for user experience. 😄 Requiring PIP for payment works for me.